sdks/python/apache_beam/transforms/external_transform_provider_test.py

```diff
@@ -407,7 +432,8 @@ def test_run_pipeline_with_script_generated_transform(self):
 
     with open(self.service_config_path, 'w') as f:
       yaml.dump([expansion_service_config], f)
-    generate_transforms_config(self.service_config_path, self.transform_config_path)
+    generate_transforms_config(
```

Comment:
why do we generate  wrappers for this test, as opposed to taking the already generated wrappers and testing them? It seems that wrappers would be generated during beam installation. reusing them might remove the test dependency on jinja/yapf.

Reply:
Good catch, we don't need to generate it for this test.

We still need jinja/yapf to test if the script handles modifications to name/destination properly (see `test_script_workflow_with_modified_transforms`)

Reply:
test_script_workflow_with_modified_transforms could potentially also be tested on generated code, or alternatively test could run only if the jinja/yapf is available, in that case we'd run the test less (manually only), but we would need less dependencies to manage.

Reply:
Added an option to not run yapf during tests.
Added an option to the xlang test utilities to supply additional dependencies. We will instead install jinja2 as an additional dependency

end result is we can remove jinja and yapf from apache_beam test dependencies